crates/duty-tracker/src/contract_manager.rs

```diff
@@ -522,8 +559,15 @@ impl ContractManagerCtx {
         let mut new_contracts = Vec::new();
 
         let pov_key = self.cfg.operator_table.pov_p2p_key().clone();
-        // The next contract will have its index at the tip of the current stake chain.
-        let deposit_idx_offset = self.state.stake_chains.height();
+        // The next contract will have its index at a value that is 1 more than the max deposit
+        // index that has already been processed.
+        let deposit_idx_offset = self
+            .state
+            .active_contracts
+            .values()
+            .map(|contract| contract.deposit_idx() + 1) // start at the max index + 1
+            .max()
```

Comment:
I think this is a more "structurally correct computation". You don't have to incorporate this change if you don't want to but it makes the provenance of this computation a little clearer

```suggestion
            .map(|contract| contract.deposit_idx())
            .max()
            .map(|max| max + 1) // start at the max index + 1
```

Reply:
Agreed. I've ammended the commit: https://github.com/alpenlabs/strata-bridge/commit/95fca683f7423ca12343663a9ebec064364ad45d.